cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -0,0 +1,202 @@
+            val confirmPayback = lenderSession
+                    .sendAndReceive<Boolean>("ready for receiving payback?")
+                    .unwrap { it }
+
+            require(confirmPayback) { "Oops, lender failed to be ready" }
+
+            progressTracker.currentStep = PAYBACK_COMMUNICATION
+            lenderSession.send(iouState.value)
+
+            progressTracker.currentStep = GATHERING_SIGS
+            val fullySignedTx = subFlow(CollectSignaturesFlow(
+                    signedTx,
+                    setOf(lenderSession, bankSession),
+                    GATHERING_SIGS.childProgressTracker())
+            )
+
+            progressTracker.currentStep = FINALISING_TRANSACTION
+            // Notarise and record the transaction in both parties' vaults.
+            return subFlow(FinalityFlow(fullySignedTx, setOf(lenderSession, bankSession), FINALISING_TRANSACTION.childProgressTracker()))
+        }
+    }
+
+    @InitiatedBy(Initiator::class)
+    class Acceptor(val borrowerPartySession: FlowSession) : FlowLogic<SignedTransaction>() {
+        @Suspendable
+        override fun call(): SignedTransaction {
+            val bank = serviceHub.bankProvider()
+
+            if (!ourIdentity.equals(bank)) {
+                borrowerPartySession.receive<String>().unwrap { it }
+                //send confirmation about being ready to receive payback
+                borrowerPartySession.send(true)
+
+                // just exercise
+                val receivedPayback: Int = borrowerPartySession.receive<Int>().unwrap { it }
+            }
+
+            val signTransactionFlow = object : SignTransactionFlow(borrowerPartySession) {
+                override fun checkTransaction(stx: SignedTransaction) = requireThat {
+                    val allInputStates = stx.tx.inputs.getStxStates(serviceHub)
+
+                    val inIOUState = allInputStates.filterByState<IOUState>().single()
+                    val inCashStates = allInputStates.filterByState<CashState>()
+
+                    val outCashStates = stx.tx.outputsOfType<CashState>()
+                    val lenderPaybackCash = outCashStates
+                            .stream().filter { it.owner.equals(inIOUState.lender) }
+                            .toList()
+
+                    val inputCashSum = inCashStates.sumByLongSecure { it.value }
+                    val lenderPaybackSum = lenderPaybackCash.sumByLongSecure { it.value }
+
+                    "Expect exact payback value. $lenderPaybackSum != ${inIOUState.value}" using (lenderPaybackSum == inIOUState.value.toLong())
+
+                    val borrowerChange = outCashStates
+                            .stream().filter { it.owner.equals(inIOUState.borrower) }
+                            .toList()
+                            .sumByLongSecure { it.value }
+                    "Expect change for borrower." using (inputCashSum - lenderPaybackSum == borrowerChange)
```

Comment:
As long as the lender is paid, it should not care about where the change went, IMO.

Reply:
good point, will add check for that.